### PR TITLE
Add interactive examples

### PR DESCRIPTION
**Pull request by bsmth**
Including these in the content repo after the interactive examples have been added:

__Related pull requests:__
- [x] https://github.com/mdn/interactive-examples/pull/2341
- [x] https://github.com/mdn/interactive-examples/pull/2343 
- [x] https://github.com/mdn/interactive-examples/pull/2345

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ browser-compat: javascript.builtins.AsyncGeneratorFunction
 
 The **`AsyncGeneratorFunction`** object provides methods for [async generator functions](/en-US/docs/Web/JavaScript/Reference/Statements/async_function*). In JavaScript, every async generator function is actually an `AsyncGeneratorFunction` object.
 
-Note that `AsyncGeneratorFunction` is _not_ a global object. It can be obtained with the following code:
+There is no global object with the name `AsyncGeneratorFunction`, but you can create a `AsyncGeneratorFunction()` constructor.
+
+{{EmbedInteractiveExample("pages/js/async-functionasterisk-function.html")}}
+
+## Syntax
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.md:15]: That's not really the syntax. Could you move the `EmbedInteractiveExample` below the code block and keep the previous intro paragraph intact?

File: files/en-us/web/javascript/reference/global_objects/generatorfunction/index.md [hunk 2]
```diff
@@ -13,7 +13,11 @@ browser-compat: javascript.builtins.GeneratorFunction
 
 {{JSRef}}
 
-In JavaScript, every generator function is actually a `GeneratorFunction` object. There is no global object with the name `GeneratorFunction`, but you can create a `GeneratorFunction()` constructor using the following code:
+In JavaScript, every generator function is actually a `GeneratorFunction` object. There is no global object with the name `GeneratorFunction`, but you can create a `GeneratorFunction()` constructor.
+
+{{EmbedInteractiveExample("pages/js/functionasterisk-function.html")}}
+
+## Syntax
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/generatorfunction/index.md:20]: Same here.

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.md]: Thanks, Josh, done in 1ed103eb62b0d6241abce3ca8e3a3b7b2308d252

File: files/en-us/web/javascript/reference/global_objects/generatorfunction/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/web/javascript/reference/global_objects/generatorfunction/index.md]: done in 1ed103eb62b0d6241abce3ca8e3a3b7b2308d252

**Review by Josh-Cena — APPROVED** [review]
Thanks!

**Comment by bsmth** [comment]
Thank you 🙌🏻

**Pull request merged**